src/main/java/util/RepoAliasMap.java

```diff
@@ -0,0 +1,260 @@
+package util;
+
+import com.google.common.collect.HashBiMap;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.*;
+import java.util.Map;
+
+/**
+ * This class acts like a bi-directional map that maps repo ids with their aliases, if the aliases exist.
+ * The mapping is a bijection, i.e. a repo id can only have one alias, and an alias can only be mapped to one repo id.
+ *
+ * This is a singleton class. Use the getInstance method to retrieve the instance.
+ * One can also construct a test instance. However this is only meant for test usage.
+ *
+ * The mapping of aliases are pulled from a mapping file in the settings directory.
+ * There is currently no functionality to update the mapping file from the code.
+ * The file has to be manually inserted
+ */
+public final class RepoAliasMap {
```

Comment:
Yeah, at least for now. Future work can be done to separate the files into one that can be safely edited, checked into version control, etc. and one that should be left alone.